Core/Core/Planner/Model/UseCase/GetPlannables+ObserverEvents.swift

```diff
@@ -0,0 +1,210 @@
+
+        guard let userID = userID else {
+            return Just([]).eraseToAnyPublisher()
+        }
+
+        let request = GetCoursesRequest(
+            enrollmentState: .active,
+            enrollmentType: .observer,
+            state: [.available],
+            perPage: 100
+        )
+
+        return Future<[Context], Never> { promise in
+
+            env.api.exhaust(request) { response, _, _ in
+                guard let courses = response else {
+                    promise(.success([]))
+                    return
+                }
+
+                var codes: [String] = []
+                for course in courses {
+                    let enrollments = course.enrollments ?? []
+                    for enrollment in enrollments where enrollment.associated_user_id?.value == userID {
+                        codes.append(Context(.course, id: course.id.value).canvasContextID)
+                    }
+                }
+
+                let contexts = codes.compactMap({ Context(canvasContextID: $0) })
+                promise(.success(contexts))
+            }
+        }
+        .eraseToAnyPublisher()
+    }
+
+    private func calendarEventsPublisher(env: AppEnvironment, contexts: [Context], for type: CalendarEventType) -> Future<[ObserverEvent], EventsFailure> {
+
+        let request = GetCalendarEventsRequest(
+            contexts: contexts,
+            startDate: startDate,
+            endDate: endDate,
+            type: type,
+            include: [.submission],
+            allEvents: false,
+            userID: userID
+        )
+
+        return Future<[ObserverEvent], EventsFailure> { promise in
+
+            env.api.exhaust(request) { response, urlResponse, error in
+
+                if let events = response?.map({ ObserverEvent.calendarEvent($0) }) {
+                    promise(.success(events))
+                } else if let error {
+                    promise(.failure(.error(error, response: urlResponse)))
+                } else {
+                    promise(.failure(.noData(given: urlResponse)))
+                }
+            }
+        }
+    }
+
+    private func plannerNotesPublisher(env: AppEnvironment, contexts: [Context]) -> Future<[ObserverEvent], EventsFailure> {
+
+        let request = GetPlannerNotesRequest(
+            contexts: contexts,
+            startDate: startDate,
+            endDate: endDate
+        )
+
+        return Future<[ObserverEvent], EventsFailure> { promise in
+
+            env.api.exhaust(request) { response, urlResponse, error in
+
+                if let notes = response?.map({ ObserverEvent.plannerNote($0) }) {
+                    promise(.success(notes))
+                } else if let error {
+                    promise(.failure(.error(error, response: urlResponse)))
+                } else {
+                    promise(.failure(.noData(given: urlResponse)))
+                }
+            }
+        }
+    }
+}
+
+// MARK: - Models
+
+enum ObserverEvent {
```

Comment:
Please either make these enums and structs private if possible or namespace them if not (via prefix or nesting)